api/routers/personalities.py

```
@@ -0,0 +1,58 @@
+    return personalities
+
+@router.get("/personalities/{personality_name}", response_model=schemas.Personality, tags=["Personalities"], dependencies=[Depends(get_current_user)])
+def get_personality_by_name(personality_name: str, db: Session = Depends(get_db)):
+    """Retorna uma personalidade da IA pelo nome."""
+    personality = personality_crud.get_personality_by_name(db, name=personality_name)
+    if personality is None:
+        raise HTTPException(status_code=404, detail="Personalidade não encontrada")
+    return personality
+
+@router.put("/personalities/{personality_name}", response_model=schemas.Personality, tags=["Personalities"], dependencies=[Depends(get_current_user)])
+def update_personality(personality_name: str, personality: schemas.PersonalityCreate, db: Session = Depends(get_db)):
```

Ruff (0.12.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)